src/test/java/de/undercouch/gradle/tasks/download/DownloadTaskPluginTest.java

```diff
@@ -214,6 +214,24 @@ public void downloadSingleURL() throws Exception {
         byte[] dstContents = FileUtils.readFileToByteArray(dst);
         assertArrayEquals(contents, dstContents);
     }
+
+    /**
+     * Tests if a single file can be downloaded from a URL
+     * only if the source file is newer
+     * @throws Exception if anything goes wrong
+     */
+    @Test
+    public void downloadSingleURLOnlyIfNewer() throws Exception {
```

Comment:
Sorry. You are right. I didn't have enough time. I had to use your library internal here in my company to cache some files. And I had to put this feature to detected corrupted/broken files. I'll add more test cases (no dest file, broken dest file, same dest file) with time-stamping and send a new pull request.